### DISCUSSION TITLE
能不能加上批量skynet.send和skynet.call的接口

### DISCUSSION THREAD

**Discussion by yuanfengyun** [cloudwu/skynet#1596] [General]
形如这种调用，明明可以一次解决的，发了两次：
    skynet.call(dbSvr, "lua", "save_auth", "auth", account, uid)
    skynet.call(dbSvr, "lua", "save", uid, {"user"}, {user = new_info,is_change = true})

**Comment by cloudwu** [comment]
两次有什么问题？

**Reply by yuanfengyun** [reply]
能省则省

**Reply by cloudwu** [reply]
为什么合并了就是省？

**Comment by JieTrancender** [comment]
不知道你是不是想要这个:
~~~
local function packMsg(cmd, ...)
    return {cmd, table.pack(...)}
end

local msgList = {}
table.insert(msgList, packMsg("save_auth", "auth", account, uid))
table.insert(msgList, packMsg("save", uid, {"user"}, {user = new_info,is_change = true}))
skynet.call(dbSvr, "lua", "dispatch", msgList)
~~~

**Comment by fanyh** [comment]
他要的是skynet.request
wiki 里面有

**Comment by JieTrancender** [comment]
如果只是单纯不想等着一个一个call，可能你需要的是这个：https://github.com/cloudwu/skynet/issues/1175#issue-601744681

**Reply by yuanfengyun** [reply]
单纯的想减少消息投递的次数，没有那么复杂的想法

**Reply by JieTrancender** [reply]
那就是我上面发那个

**Reply by yuanfengyun** [reply]
我说的是批量投递到同一个服务，不是等待多个服务同时返回。不是同一个问题。

**Reply by JieTrancender** [reply]
上面发那个指这个：https://github.com/cloudwu/skynet/discussions/1596#discussioncomment-2736063

**Reply by SheepQi** [reply]
这个自己实现有很多中方法的嘛，比如你可以自己定义一类消息比如叫patch,然后在skynet.dispatch()里面去处理对patch消息的拆包。又或者你直接在对应的服务里面加一个patch_call和patch_send方法去处理。
